### DISCUSSION TITLE
Should we fork `async-session`?

### DISCUSSION THREAD

**Discussion by maxcountryman** [maxcountryman/axum-sessions#23] [General]
`async-session` was last released in June, 2021. There is [one pending PR](https://github.com/http-rs/async-session/pull/14) that's been open for two years.

The maintainer has said the crate is maintained however it's unclear what that commitment means and whether there's interest in supporting the crate beyond its original target: the Tide web framework. (Based on some prior discussion around the project, it seems there's stated interest, but in practice I'm struggling to find continued movement around that.)

There are also certain design decisions which have led to less-than-ideal ergonomics in this crate ([see](https://github.com/maxcountryman/axum-sessions/discussions/20)).

Generally I would like to avoid forking and fragmenting the ecosystem. That said, this may be an instance where it's worth the investment and if done carefully perhaps can be eventually merged back into the upstream project.

I'd love any thoughts or feedback.

**Comment by LeoniePhiline** [comment]
Sounds like you managed to get some attention from the maintainer - good job!

The next right thing, as it seems to be, would be to open an issue regarding their design decisions and the issues it creates here. They might agree that breaking changes to the design to improve the situation could be part of 4.0. 

Depending on their answer we can decide if to base an improved design on their original code or to contribute upstream.

**Comment by maxcountryman** [comment]
I believe there's material changes coming to the upstream crate so I'm strongly in favor of doubling down on supporting that and not trying to maintain a separate fork.

**Reply by LeoniePhiline** [reply]
https://github.com/http-rs/async-session/pull/57 looks really good. 
Thanks for having an eye on it and reminding the maintainer of the clonability issue!
